schema/handler.go

```diff
@@ -77,34 +80,137 @@ type getJsonSchema struct {
+		return
+	}
+	defer src.Close()
+
+	w.Header().Add("Content-Type", "application/json")
+	if _, err := io.Copy(w, src); err != nil {
+		h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The file for this JSON Schema ID could not be found or opened. This is a configuration issue.").WithDebugf("%+v", err)))
+		return
+	}
+}
+
+// Raw identity Schema list
+//
+// swagger:model IdentitySchemas
+type IdentitySchemas []identitySchema
+
+// swagger:model identitySchema
+type identitySchema struct {
+	// The ID of the Identity JSON Schema
+	ID string `json:"id"`
+	// The actual Identity JSON Schema
+	Schema json.RawMessage `json:"schema"`
+}
+
+// nolint:deadcode,unused
+// swagger:parameters listIdentitySchemas
+type listIdentitySchemas struct {
+	// Items per Page
+	//
+	// This is the number of items per page.
+	//
+	// required: false
+	// in: query
+	// default: 100
+	// min: 1
+	// max: 500
+	PerPage int `json:"per_page"`
+
+	// Pagination Page
+	//
+	// required: false
+	// in: query
+	// default: 0
+	// min: 0
+	Page int `json:"page"`
+}
+
+// swagger:route GET /schemas v0alpha2 listIdentitySchemas
+//
+// Get all Identity Schemas
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       200: IdentitySchemas
+//       500: jsonError
+func (h *Handler) getAll(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	page, itemsPerPage := x.ParsePagination(r)
+
+	schemas := h.r.IdentityTraitsSchemas(r.Context()).List(page, itemsPerPage)
+	total := h.r.IdentityTraitsSchemas(r.Context()).Total()
+
+	var ss IdentitySchemas
+
+	for _, schema := range schemas {
+		s, err := h.r.IdentityTraitsSchemas(r.Context()).GetByID(schema.ID)
+		if err != nil {
+			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrNotFound.WithDebugf("%+v", err)))
+			return
+		}
+
+		src, err := ReadSchema(s)
 		if err != nil {
 			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The file for this JSON Schema ID could not be found or opened. This is a configuration issue.").WithDebugf("%+v", err)))
 			return
 		}
 		defer src.Close()
-	} else {
-		resp, err := http.Get(s.URL.String())
+
+		raw, err := ioutil.ReadAll(src)
 		if err != nil {
 			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The file for this JSON Schema ID could not be found or opened. This is a configuration issue.").WithDebugf("%+v", err)))
 			return
 		}
-		defer resp.Body.Close()
-		src = resp.Body
+
+		var p json.RawMessage
+		err = json.Unmarshal(raw, &p)
+		if err != nil {
+			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The file for this JSON Schema ID could not be found or opened. This is a configuration issue.").WithDebugf("%+v", err)))
+			return
+		}
```

Comment:
You don’t need to unmarshall. `json.RawMessage` is just `[]byte` so no need to decode here, just assign `Schema: raw` :)